services/visual_recognition/v3.js

```diff
@@ -0,0 +1,348 @@
+          message = 'Bad format : msg.payload must be a URL string or a Node.js Buffer';
+          node.error(message, msg);
+          return;
+        }
+
+      // If it is present the newly provided user entered key takes precedence over the existing one. 
+      apikey = s_apikey || this.credentials.apikey;
+      this.status({}); 
+
+      if (!apikey) {
+        this.status({fill:'red', shape:'ring', text:'missing credentials'});          
+        message ='Missing Watson Visual Recognition API service credentials'; 
+        node.error(message, msg);
+        return;
+      }
+
+      var params = {};
+
+      // This is the callback after the call to the watson service.    
+      // Set up as a var within this scope, so it has access to node, msg etc. 
+      // in preparation for the Watson service action       
+      var actionComplete = function(err, body, other) {
+
+        if (err != null && body==null)
+        {
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_code']= err.code;
+          if (!err.error)
+            msg.result['error']= err.error;
+          console.log('Error:', err.code);
+          node.error('Error code : ' +  err.code);
+          return;
+        }
+        else if (body.images[0].error)
+        {
+          var err_desc = body.images[0].error.description;
+          var err_id = body.images[0].error.error_id
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_id']= err_id;
+          msg.result['error']= err_desc;
+          console.log('Error:', err_desc);
+          msg.payload='see msg.result.error';
+          node.send(msg); 
+        }
+        else {
+          msg.result = {};
+          msg.result['all'] = body;
+          msg.payload='see msg.result'; // to remove any Buffer that could remains
+          node.send(msg); 
+          node.status({});
+        }
+      } // actionComplete    
+
+      var actionComplete2 = function(err, body, other) {
+
+        if (err != null && body==null)
+        {
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
+          msg.result = {};
+          msg.result['error_code']= err.code;
+          if (!err.error)
+            msg.result['error']= err.error;
+          console.log('Error:', err.code);
+          node.error('Error code : ' +  err.code);
+          return;
+        }
+        else {
+          msg.result = {};
+          msg.result['all'] = body;
+          msg.payload='see msg.result'; // to remove any Buffer that could remains
+          node.send(msg); 
+          node.status({});
+        }
+      } // actionComplete2
+
+      var actionCompleteDeleteClassifier = function(err, body, other) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['other' is defined but never used.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2686950655/issues/source?bid=3225115&fileBranchId=3338101#l214)